### PR TITLE
Fixes #14804 - Activation keys filter by content view

### PR DESCRIPTION
**Pull request by johnpmitsch**
It looks like we weren't finding the content view before, but still using a @content_view global variable. I added a function to find the content view on the index call for activation keys. I added a test this functionality as well (it is failing on master)

### CODE REVIEW

**Comment by johnpmitsch** [comment]
To test: Make sure activation keys are sorted by the content view selected when creating a new hostgroup. Also, check that activation keys are listed normally elsewhere in the UI

File: app/controllers/katello/api/v2/activation_keys_controller.rb [hunk 1]
```diff
@@ -293,6 +294,10 @@ def verify_presence_of_organization_or_environment
       fail HttpErrors::BadRequest, _("Either organization ID or environment ID needs to be specified")
     end
 
+    def find_content_view
+      @content_view = ContentView.find_by(:id => params[:content_view_id])
```
Comment (ehelms) [inline app/controllers/katello/api/v2/activation_keys_controller.rb]: This should probably throw an error like [1] if no content_view found.

[1] https://github.com/Katello/katello/pull/5995/files#diff-65d2a2cbc038a622df645394a941f518R286

**Comment by johnpmitsch** [comment]
@ehelms updated

**Comment by zjhuntin** [comment]
Tested, LGTM ACK

**Pull request merged**